### ISSUE TITLE
Workspace import doesnt honor mm/s vs mm/min setting

### ISSUE THREAD

**Issue by openhardwarecoza** [LaserWeb/LaserWeb4#284]
### Follow this steps to open an issue ###

- [x] If refers software, include `Workspace.json`.  This is available on *CAM tab top, Workspace -> Save*. Prepare your work, save `workspace.json`, rename as `workspace.json.txt` and attach in this issue. *All resources regarding the issue should be attached (SVG files, images, etc.). An issue opened without resources will be automatically closed upon inspection*

[4040foot.zip](https://github.com/LaserWeb/LaserWeb4/files/951032/4040foot.zip) (workspace)
[4040foot dxf.zip](https://github.com/LaserWeb/LaserWeb4/files/951037/4040foot.dxf.zip) (dxf)

I created this workspace on my home PC where settings are on mm/sec 

Took it to PC in the garage, which, i didnt notice, was on mm/min (; 

Loading the workspace, thus loaded my 150mm/sec as 150mm/min - and since I didnt notice, i merilly generated gcode, and ran... luckily /sec to /min results in a machine moving very much slower than planned - had it been the other way round (house pc on mm/min, pc next to machine on mm/sec) i wouldve needed a new endmill and spare underpants (; 

I guess best practice it should probably always export in mm/min (and be changed to mm/sec if loaded with that setting enabled) or something like that (;

**Comment by jorgerobles** [comment]
@openhardwarecoza workspace snapshot captures 'documents', 'operations', 'currentOperation' states to download. Could include `settings` but that could lead unwanted behaviour, as is not strictly `workspace`  😏 
Other way to achieve that is using tools application snapshot that makes a copy of the specified state keys.  Being "mm/s" setting critical, could try add subkeys saving and add it to workspace snapshot ('settings.toolFeedUnits')

**Comment by openhardwarecoza** [comment]
Or perhaps we could always capture as a known unit (mm/min as thats what we
send the firmware anyway) , and convert everywhere its displayed as mm/s?
Ie not a change to workspace export, just a change to the way values are
handled? (mm/min under the hood, /60 whenever it needs to be?

**Comment by jorgerobles** [comment]
.. :smile: ugh, maybe lot of work that way. Have to search whole application for the use of that setting in each different team implementation. Wanna help?

**Comment by openhardwarecoza** [comment]
Hehe Yeah leave it open for now will do soon! Its not a majorly urgent one

**Comment by jorgerobles** [comment]
We should ditch mm/s and done 😆

**Comment by openhardwarecoza** [comment]
Lol i kinda agree. Its the corel/laserdrw crowd that didnt want to

**Comment by openhardwarecoza** [comment]
We can always force it on them

**Comment by FabCreator** [comment]
Or ditch mm/min and so for SI units only ;)

**Comment by cprezzi** [comment]
For me it's easyer to imagine mm/s, but it looks like there is no real industry standard (see https://en.wikipedia.org/wiki/Speeds_and_feeds).

**Comment by openhardwarecoza** [comment]
(: the Mericans want inch per minute and G20 support (;

**Comment by cprezzi** [comment]
Not all of them ;) The US industry is slowly moving :)

**Comment by openhardwarecoza** [comment]
(; ok let me re-stereotype lol. The old ones lol

**Comment by FabCreator** [comment]
I agree with @cprezzi i find mm/s easier to imagine, too many zeros for mm/min. 100mm/s vs 6000mm/m
Lets do the Americans a favour and leave out imperial from laserweb ;)

**Comment by jorgerobles** [comment]
With patch #284, workspace snapshot (which is indeed a part of the whole application state) stores also settings.toolFeedUnits. So, if you load that workspace, that setting will overwrite config. That fixes this issue, but caveats are that as is done, that setting change is permanent upon workspace load.

An alternative way is to store the units with the value (Ex: "100 mm/s" ) then the loader and inputs have to resolve (regexp) this kind of notation. That's an UI challenge then.

**Comment by tbfleming** [comment]
jscut stored units with values; it created a large source of bugs that I had to fix. It also forced me to roll back a pull request I accepted a few months later because that pull request interacted subtly with units creating a hard-to-reproduce bug that a few key users kept hitting. Units with values did have a user advantage though; users could enter imperial tool units and metric operation units without having to do the conversion on their own.
